boards/arm/b_g474e_dpow1/board.c

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2022 The Chromium OS Authors
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/zephyr.h>
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/adc.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(board, LOG_LEVEL_DBG);
+
+#if DT_HAS_COMPAT_STATUS_OKAY(usbc_port)
```

Comment:
this can be avoided if this code is part of a sample, and sample provides a proper filter. for example, you can add to `sample.yaml`:

```yaml
filter: dt_compat_enabled("usbc-port")
```

Reply:
I thought sample.yaml was used during testing. Could you give a more complete example of how "filter" could be used to exclude board.c when the usbc-port is not defined?